kernel/include/priority_q.h

```diff
@@ -279,23 +302,21 @@ static ALWAYS_INLINE void z_priq_mq_yield(struct _priq_mq *pq)
 
 static ALWAYS_INLINE struct k_thread *z_priq_mq_best(struct _priq_mq *pq)
 {
-	struct k_thread *thread = NULL;
-
-	for (int i = 0; i < PRIQ_BITMAP_SIZE; ++i) {
-		if (!pq->bitmask[i]) {
-			continue;
-		}
+#ifdef CONFIG_SMP
+	unsigned int index = z_priq_mq_best_prio_index(pq);
+#else
+	unsigned int index = pq->prio_index;
+#endif
 
-		sys_dlist_t *l = &pq->queues[i * NBITS + TRAILING_ZEROS(pq->bitmask[i])];
-		sys_dnode_t *n = sys_dlist_peek_head(l);
+	if (likely(index != 0xFFFFFFFF)) {
```

Comment:
Good idea. The initial state though should probably be `K_NUM_THREAD_PRIO - 1`--the slot for the lowest priority thread so that the update works correctly.

This should also reduce the patch set by 1. :)